### PR TITLE
Add `GetJobs`

### PR DESCRIPTION
**Pull request by quinthar**
This adds a plural `GetJobs` command that returns an array of jobs, all in one call.  The singular `GetJob` command is retained unchanged for backwards compatibility.

Tests:
----
Here is a test session, walking it through its paces:

```
vagrant@vagrant-ubuntu-trusty-64:/vagrant/Bedrock$ nc localhost 8888
GetJob   
name: asdf

404 No job found
commitCount: 1
nodeName: vagrant-ubuntu-trusty-64
processingTime: 3
totalTime: 3
waitTime: 0
Content-Length: 0

GetJobs
name: asdf

402 Missing numResults
commitCount: 1
nodeName: vagrant-ubuntu-trusty-64
processingTime: 0
totalTime: 0
waitTime: 0
Content-Length: 0

GetJobs
name: asdf
numResults: 10

404 No job found
commitCount: 1
nodeName: vagrant-ubuntu-trusty-64
processingTime: 0
totalTime: 1
waitTime: 0
Content-Length: 0

CreateJob
name: asdf

200 OK
commitCount: 2
nodeName: vagrant-ubuntu-trusty-64
processingTime: 2
totalTime: 3
waitTime: 1
Content-Length: 11

{"jobID":1}

GetJob
name: asdf

200 OK
commitCount: 2
nodeName: vagrant-ubuntu-trusty-64
processingTime: 3
totalTime: 5
waitTime: 2
Content-Length: 35

{"data":{},"jobID":1,"name":"asdf"}

RetryJob
jobID: 1

200 OK
commitCount: 3
nodeName: vagrant-ubuntu-trusty-64
processingTime: 0
totalTime: 3
waitTime: 2
Content-Length: 0

GetJobs
name: asdf
numResults: 10

200 OK
commitCount: 4
nodeName: vagrant-ubuntu-trusty-64
processingTime: 6
totalTime: 8
waitTime: 2
Content-Length: 46

{"jobs":[{"data":{},"jobID":1,"name":"asdf"}]}

CreateJob
name: parent
data: {"foo":1} 

200 OK
commitCount: 5
nodeName: vagrant-ubuntu-trusty-64
processingTime: 1
totalTime: 3
waitTime: 2
Content-Length: 11

{"jobID":2}

CreateJob
name: child
data: {"bar":2} 
parentJobID: 2

200 OK
commitCount: 6
nodeName: vagrant-ubuntu-trusty-64
processingTime: 1
totalTime: 3
waitTime: 1
Content-Length: 11

{"jobID":3}

query: select * from jobs;

200 OK
commitCount: 7
nodeName: vagrant-ubuntu-trusty-64
processingTime: 0
totalTime: 1
waitTime: 1
Content-Length: 372

created | jobID | state | name | nextRun | lastRun | repeat | data | priority | parentJobID
2017-01-29 21:30:47 | 1 | RUNNING | asdf | 2017-01-29 21:31:01 | 2017-01-29 21:31:07 |  | {} | 0 | 0
2017-01-29 21:31:30 | 2 | QUEUED | parent | 2017-01-29 21:31:30 |  |  | {"foo":1} | 0 | 0
2017-01-29 21:31:43 | 3 | QUEUED | child | 2017-01-29 21:31:43 |  |  | {"bar":2} | 0 | 2


GetJobs
name: *
numResults: 10

200 OK
commitCount: 7
nodeName: vagrant-ubuntu-trusty-64
processingTime: 3
totalTime: 6
waitTime: 2
Content-Length: 138

{"jobs":[{"data":{"foo":1},"jobID":2,"name":"parent"},{"data":{"bar":2},"jobID":3,"name":"child","parentData":{"foo":1},"parentJobID":2}]}

GetJob
connection: wait
name: qwer

200 OK
commitCount: 8
nodeName: vagrant-ubuntu-trusty-64
processingTime: 6
totalTime: 36562
waitTime: 36556
Content-Length: 35

{"data":{},"jobID":4,"name":"qwer"}

GetJobs
connection: wait
name: zxcv

402 Missing numResults
commitCount: 10
nodeName: vagrant-ubuntu-trusty-64
processingTime: 0
totalTime: 0
waitTime: 0
Content-Length: 0

GetJobs
connection: wait
name: zxcv
numResults: 10

200 OK
commitCount: 10
nodeName: vagrant-ubuntu-trusty-64
processingTime: 5
totalTime: 11305
waitTime: 11299
Content-Length: 46

{"jobs":[{"data":{},"jobID":5,"name":"zxcv"}]}
```

Cc: @tylerkaraszewski @mcnamamj

### CODE REVIEW

File: plugins/Jobs.cpp [hunk 1]
```diff
@@ -280,36 +295,52 @@ bool BedrockPlugin_Jobs::processCommand(BedrockNode* node, SQLite& db, BedrockNo
...
     // ----------------------------------------------------------------------
-    else if (SIEquals(request.methodLine, "GetJob")) {
-        // If we're here it's because peekCommand found some data;
-        // re-execute the query for real now.
+    else if (SIEquals(request.methodLine, "GetJob") || SIEquals(request.methodLine, "GetJobs")) {
+        // If we're here it's because peekCommand found some data; re-execute
+        // the query for real now.  However, this time we will order by
+        // priority.  We do this as three separate queries so we only have one
+        // unbounded column in each query.  Additionally, we wrap each inner
+        // query in a "SELECT *" such that we can have an "ORDER BY" and
```
Comment (cead22) [inline plugins/Jobs.cpp]: Is this part really necessary? Aren't these equivalent?
```sql
SELECT jobID, name, data, parentJobID FROM ( 
    SELECT * FROM (
        SELECT jobID, name, data, priority, parentJobID
         ...
        ORDER BY nextRun ASC LIMIT safeNumResults
    )
    UNION ALL 
    SELECT * FROM (
        SELECT jobID, name, data, priority, parentJobID
        ... 
        ORDER BY nextRun ASC LIMIT safeNumResults
    )
...
)
```

```sql
SELECT jobID, name, data, parentJobID FROM (

    SELECT jobID, name, data, priority, parentJobID
     ...
    ORDER BY nextRun ASC LIMIT safeNumResults
    UNION ALL 
    
    SELECT jobID, name, data, priority, parentJobID
    ... 
    ORDER BY nextRun ASC LIMIT safeNumResults
     ...
)
```

File: plugins/Jobs.cpp [hunk 2]
```diff
@@ -658,18 +712,17 @@ string BedrockPlugin_Jobs::_constructNextRunDATETIME(const string& lastScheduled
     return "DATETIME( " + SComposeList(safeParts) + " )";
 }
 
-/**
- * Returns true if there are any children of this jobID in a RUNNING or QUEUED state.
- */
+// ==========================================================================
```
Comment (cead22) [inline plugins/Jobs.cpp]: My least favorite thing. Not a blocker.

File: plugins/Jobs.cpp [hunk 3]
```diff
@@ -658,18 +712,17 @@ string BedrockPlugin_Jobs::_constructNextRunDATETIME(const string& lastScheduled
...
 bool BedrockPlugin_Jobs::_hasPendingChildJobs(SQLite& db, int64_t jobID) {
+    // Returns true if there are any children of this jobID in a RUNNING or QUEUED state.
```
Comment (cead22) [inline plugins/Jobs.cpp]: Update to include PAUSED state

File: plugins/Jobs.cpp (hunk 1 above)
Comment (quinthar) [inline plugins/Jobs.cpp]: The `ORDER BY` needs to come *after* the `UNION ALL`, which is why I wrapped each of the internal queries in a subselect and then unioned them together.  For example:

```
sqlite> create table blah ( value );
sqlite> select * from blah UNION ALL select * from blah;
sqlite> 
sqlite> select * from blah UNION ALL select * from blah order by value desc;
sqlite> 
sqlite> select * from blah order by value desc UNION ALL select * from blah;
Error: ORDER BY clause should come after UNION ALL not before
sqlite>
```

**Comment by quinthar** [comment]
Addresses all blocking comments from @cead22

**Pull request merged**